### PR TITLE
feat: add cluster visibility to FalconAdmission

### PR DESCRIPTION
**Pull request by gpontejos**

### CODE REVIEW

File: api/falcon/v1alpha1/falconadmission_types.go [hunk 1]
```diff
@@ -201,3 +229,35 @@ type FalconAdmissionList struct {
...
+	return watcher.SnapshotsInterval.Duration.String()
+}
+
+func (watcher FalconAdmissionConfigSpec) GetWatcherEnabled() string {
+	if watcher.WatcherEnabled == nil {
+		return "true"
+	}
+
+	return strconv.FormatBool(*watcher.WatcherEnabled)
+}
```
Comment (mhyson-cs) [inline api/falcon/v1alpha1/falconadmission_types.go:263]: These functions would be more generally useful if they returned their native types (bool, Duration, etc). Places in the code that need them as strings can worry about that themselves.

File: internal/controller/assets/deployment.go [hunk 2]
```diff
@@ -741,3 +849,49 @@ func admissionDepUpdateStrategy(admission *falconv1alpha1.FalconAdmission) appsv
...
+			Value: admission.Spec.AdmissionConfig.GetSnapshotsEnabled(),
+		},
+		corev1.EnvVar{
+			Name:  "__CS_SNAPSHOT_INTERVAL",
+			Value: admission.Spec.AdmissionConfig.GetSnapshotsInterval(),
+		},
+		corev1.EnvVar{
+			Name:  "__CS_WATCH_EVENTS_ENABLED",
+			Value: admission.Spec.AdmissionConfig.GetWatcherEnabled(),
+		},
```
Comment (mhyson-cs) [inline internal/controller/assets/deployment.go:893]: The other side of the comment above, string-formatting of API properties is better done here.

**Review by mhyson-cs — CHANGES_REQUESTED** [review]

File: api/falcon/v1alpha1/falconadmission_types.go (hunk 1 above)
Comment (gpontejos) [reply api/falcon/v1alpha1/falconadmission_types.go]: Updated.

File: internal/controller/assets/deployment.go (hunk 2 above)
Comment (gpontejos) [reply internal/controller/assets/deployment.go]: Updated

**Review by mhyson-cs — CHANGES_REQUESTED** [review]

File: api/falcon/v1alpha1/falconadmission_types.go [hunk 3]
```diff
@@ -201,3 +229,36 @@ type FalconAdmissionList struct {
...
+		return true
+	}
+
+	return *watcher.SnapshotsEnabled
+}
+
+func (watcher FalconAdmissionConfigSpec) GetSnapshotsInterval() time.Duration {
+	if watcher.SnapshotsInterval == nil {
+		defaultSnapshotInterval, _ := time.ParseDuration("22h0m0s")
+		return defaultSnapshotInterval
```
[the remark this answers is not in the document]
Comment (gpontejos) [reply api/falcon/v1alpha1/falconadmission_types.go]: Updated

**Review by mhyson-cs — APPROVED** [review]

**Pull request merged**